File: src/main/java/com/dogGetDrunk/meetjyou/common/exception/business/DuplicateException.java
package com.dogGetDrunk.meetjyou.common.exception.business;

import com.dogGetDrunk.meetjyou.common.exception.ErrorCode;
import lombok.Getter;

@Getter
public class DuplicateException extends BusinessException {

    private String value;

    public DuplicateException(String value) {
        this(value, ErrorCode.DUPLICATE);
        this.value = value;
    }

    public DuplicateException(String value, ErrorCode errorCode) {
        super(value, errorCode);
        this.value = value;
    }
}

File: src/main/java/com/dogGetDrunk/meetjyou/common/exception/business/DuplicateException.kt
package com.dogGetDrunk.meetjyou.common.exception.business

import com.dogGetDrunk.meetjyou.common.exception.ErrorCode

open class DuplicateException(
    errorCode: ErrorCode = ErrorCode.DUPLICATE,
    value: String,
    message: String? = null,
) : BusinessException(value, errorCode, message)
